File: mobile/src/main/java/uk/co/sentinelweb/bitwatcher/common/ui/picker/PickerPresenterFactory.java
package uk.co.sentinelweb.bitwatcher.common.ui.picker;

import android.support.annotation.NonNull;

import javax.inject.Inject;

public class PickerPresenterFactory {

    @Inject
    public PickerPresenterFactory() {
    }

    public @NonNull <T> PickerContract.Presenter<T> createPresenter(PickerContract.View<T> pickerView) {
        PickerPresenter<T> pickerPresenter = new PickerPresenter<T>(pickerView);
        pickerView.setPresenter(pickerPresenter);
        return pickerPresenter;

    }
}

File: mobile/src/main/java/uk/co/sentinelweb/bitwatcher/common/ui/picker/PickerPresenterFactory.kt
package uk.co.sentinelweb.bitwatcher.common.ui.picker

import javax.inject.Inject

class PickerPresenterFactory @Inject constructor() {

    fun <T> createPresenter(pickerView: PickerContract.View<T>): PickerContract.Presenter<T> {
        val pickerPresenter = PickerPresenter(pickerView)
        pickerView.setPresenter(pickerPresenter)
        return pickerPresenter

    }
}
